mne_sandbox/connectivity/cfc.py

```diff
@@ -121,17 +121,129 @@ def phase_amplitude_coupling(inst, f_phase, f_amp, ixs, pac_func='ozkurt',
+        If `events` is provided, it is the time (in seconds) to include after
+        each event index. If a list of floats is given, then PAC is calculated
+        for each pair of `tmin` and `tmax`. Defaults to `max(inst.n_times)`.
+    n_cycles_ph : float, int | array of floats, shape (n_bands_phase,)
+        The number of cycles to be included in the window for each band-pass
+        filter for phase. Defaults to 3.
+    n_cycles_am : float, int | array of floats, shape (n_bands_amp,)
+        The number of cycles to be included in the window for each band-pass
+        filter for amplitude. Defaults to 3.
+    scale_amp_func : None | function
+        If not None, will be called on each amplitude signal in order to scale
+        the values. Function must accept an N-D input and will operate on the
+        last dimension. E.g., `sklearn.preprocessing.scale`.
+        Defaults to no scaling.
+    n_jobs : int
+        Number of jobs to run in parallel. Defaults to 1.
+    """
+    def __init__(self, f_phase, f_amp, ixs, sfreq, events=None, times=None,
+                 pac_func='ozkurt', tmin=0., tmax=1., n_cycles_ph=3,
+                 n_cycles_am=3, scale_amp_func=None, n_jobs=1):
+        if self.events is not None and self.times is not None:
+            raise ValueError('Must supply one of events and times, not both')
+        self.f_phase = f_phase
+        self.f_amp = f_amp
+        self.ixs = ixs
+        self.sfreq = sfreq
+        self.pac_func = pac_func
+        self.events = events
+        self.times = times
+        self.tmin = tmin
+        self.tmax = tmax
+        self.n_cycles_ph = n_cycles_ph
+        self.n_cycles_am = n_cycles_am
+        self.scale_amp_func = scale_amp_func
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y=None):
+        """Estimate PAC values.
+
+        Parameters
+        ----------
+        X : array, shape (n_channels, n_times) |
+                (n_epochs, n_channels, n_times)
+            The input data.
+        """
+        if self.events is not None:
+            if X.ndim == 3:
+                raise ValueError('If Events is not None, X must be 2D')
+            if self.events[:, 0].max() > X.shape[0]:
+                raise ValueError('Event index exceeds dimensions of X.')
+        if X.ndim == 2:
+            # Add an extra first dimension for the loop
+            X = [X]
+
+        pac = []
+        for ep_X in X:
+            # Output will be shape (n_epochs, n_ch_pair, n_fr_pair, n_ti_win)
+            i_pac = _phase_amplitude_coupling(
+                ep_X, self.sfreq, self.f_phase, self.f_amp, self.ixs,
+                self.pac_func, self.events, self.tmin, self.tmax, self.times,
+                self.n_cycles_ph, self.n_cycles_am, self.scale_amp_func)
+            # Reshape PAC so it's 2D and first dimension is Epochs
+            pac.append(i_pac.reshape(pac.shape[0], -1))
+        self.pac = np.asarray(pac)
+
+    def transform(X=None, y=None, self):
```

Comment:
If we want to have raws with events, events must be passed here. Although adding ad hoc param is sklearn compatible it is currently not pipelineable, so I suggest to keep this for later.

In other words, let's not support raw data for now.